### PR TITLE
BloggerApp: Validations, Models and N+1 problems

### PR DESCRIPTION
**Pull request by BenjaminSemah**
### In this branch, I have done the following
✅  Added validations for the `User` and `Post` models
✅  Added unit specs for models' methods and validations
✅ Add flash messages in the create actions in all controllers
✅ Ensured `N+1 problem` is solved
✅ Fixed linter errors

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -4,7 +4,6 @@ class PostsController < ApplicationController
 
   def index
     @user = User.find_by(id: params[:user_id])
-    @all_posts = @user.posts
   end
```
Comment (rukundoeric) [inline app/controllers/posts_controller.rb]: - [x] <b>REQUIRED:</b> why did you delete `all_posts` here, Because we will need to get all posts at some point.  and also, you will have to avoid N+1 Problem by doing something like this `@all_posts = @user.posts.includes(:comments)`.

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <% @all_posts.each do |post| %>
     <%= render partial: "shared/post", locals: {post: post} %>
```
Comment (rukundoeric) [inline app/views/posts/index.html.erb:7]: - [x] <b>REQUIRED:</b> According to afformentioned issues, Now here your app is throwing `@all_posts is undefined` error.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @BenjaminSemah, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (BenjaminSemah) [reply app/views/posts/index.html.erb:7]: ✅ Resolved.

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (BenjaminSemah) [reply app/controllers/posts_controller.rb]: ✅ Resolved.

**Review by Emmyn5600 — APPROVED** [review]
Hi @BenjaminSemah 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [ ] N/A.

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by BenjaminSemah** [comment]
Thank you @Emmyn5600
